js.go

```diff
@@ -810,7 +839,15 @@ type nextRequest struct {
 
 // jsSub includes JetStream subscription info.
 type jsSub struct {
-	js       *js
+	js *js
+
+	// To setup request mux handler for pull subscribers.
+	mu    sync.RWMutex
```

Comment:
I don't think we want a mux here. We want one per pull subscriber. Otherwise we will have issues.